activerecord/lib/active_record/migration.rb

```diff
@@ -1376,18 +1376,25 @@ def use_advisory_lock?
 
       def with_advisory_lock
         lock_id = generate_migrator_advisory_lock_id
-        AdvisoryLockBase.establish_connection(ActiveRecord::Base.connection_db_config) unless AdvisoryLockBase.connected?
-        connection = AdvisoryLockBase.connection
-        got_lock = connection.get_advisory_lock(lock_id)
-        raise ConcurrentMigrationError unless got_lock
-        load_migrated # reload schema_migrations to be sure it wasn't changed by another process before we got the lock
-        yield
-      ensure
-        if got_lock && !connection.release_advisory_lock(lock_id)
-          raise ConcurrentMigrationError.new(
-            ConcurrentMigrationError::RELEASE_LOCK_FAILED_MESSAGE
+
+        @advisory_lock_pool ||= begin
+          ActiveRecord::ConnectionAdapters::ConnectionHandler.new.establish_connection(
+            ActiveRecord::Base.connection_db_config
           )
```

Comment:
What does the memoization get us here? I'm not sure it's necessary since we only take a lock out once per migration command.

Reply:
That is a good point, I looked through the code and we indeed only take a lock once per migration. Updated the PR :+1: